src/dlgPackageExporter.cpp

```diff
@@ -662,228 +665,231 @@ void dlgPackageExporter::slot_export_package()
-                QString directoryName = itDirectoryName.next();
-                // zip_dir_add(...) returns the index of the
-                // added directory item in the archive or -1 on error:
-                if (zip_dir_add(archive, directoryName.toStdString().c_str(), ZIP_FL_ENC_UTF_8) == -1) {
-                    displayResultMessage(tr("Failed to add directory \"%1\" to package. Error is: \"%2\".")
-                                         .arg(directoryName, zip_strerror(archive)), false);
-                    zip_close(archive);
-                    isOk = false;
-                }
-            }
+    QStringListIterator itDirectoryName(directoryEntries);
+    while (itDirectoryName.hasNext() && isOk) {
+        QString directoryName = itDirectoryName.next();
+        // zip_dir_add(...) returns the index of the
+        // added directory item in the archive or -1 on error:
+        if (zip_dir_add(archive, directoryName.toStdString().c_str(), ZIP_FL_ENC_UTF_8) == -1) {
+            QString errorMsg = tr("Failed to add directory \"%1\" to package. Error is: \"%2\".")
+                                 .arg(directoryName, zip_strerror(archive));
+            zip_close(archive);
+            return {false, errorMsg};
+        }
+    }
 
-            // Process the config and the file containing the Mudlet triggers,
-            // etc. specially so they are inserted first and last respectively:
-            if (isOk) {
-                // Apparently it is permissable for there NOT to be a config.lua
-                // file in the manufactured module (i.e. the user is allowed to
-                // manually remove it even though we now request they do not).
-                // If we enhance modules in the future to store more data in
-                // that file I think this will no longer be permissable and the
-                // two commented out bits of code can be restored
-                if (fileEntries.contains(QStringLiteral("config.lua"))) {
-                    if (!writeFileToZip(QStringLiteral("config.lua"), fileEntries.value(QStringLiteral("config.lua")), archive)) {
-                        /* isOk = false; */
-                    } else {
-                        fileEntries.remove(QStringLiteral("config.lua"));
-                    }
-                }/* else {
-                    displayResultMessage(tr("Required \"config.lua\" file not found to include in the package. Did you remove or rename it?"), false);
-                    zip_close(archive);
-                    isOk = false;
-                }*/
+    // Process the config and the file containing the Mudlet triggers,
+    // etc. specially so they are inserted first and last respectively:
+    if (isOk) {
+        if (fileEntries.contains(QStringLiteral("config.lua"))) {
+            if (!writeFileToZip(QStringLiteral("config.lua"), fileEntries.value(QStringLiteral("config.lua")), archive).first) {
+                /* isOk = false; */
```

Comment:
We are not bailing out if the meta-information file cannot be put into the archive then...?